lib/groups.js

```diff
@@ -139,12 +179,54 @@ module.exports.createGroup = function (groupName, assessmentId, userId, authnUse
 };
 
 module.exports.leaveGroup = function (assessmentId, userId, authnUserId, callback) {
+  const params = [assessmentId, userId, authnUserId];
+  sqldb.call('group_leave', params, function (err) {
+    if (ERR(err, callback)) return;
+    callback();
+  });
+};
+
+module.exports.getAssessmentLevelPermissions = function (assessmentId, userId, callback) {
   const params = {
     assessment_id: assessmentId,
     user_id: userId,
-    authn_user_id: authnUserId,
   };
-  sqldb.query(sql.leave_group, params, function (err) {
+  sqldb.query(sql.get_assessment_level_permissions, params, function (err, result) {
+    if (ERR(err, callback)) return;
+    callback(result.rows[0]);
+  });
+};
+
+module.exports.updateGroupRoles = function (
+  requestBody,
+  assessmentId,
+  userId,
+  authnUserId,
+  callback
+) {
+  // Convert form data to valid input format for a SQL function
+  const roleKeys = Object.keys(requestBody).filter((key) => !key.startsWith('__'));
+  const uidToRoleIdMap = {};
+  for (const roleKey of roleKeys) {
+    const [roleId, uid] = roleKey.split('-');
```

Comment:
Is the thing after the first `-` really a UID? If so, this isn't safe - what if a UID itself contains a `-`? I'd recommend using the `user_id` instead.

Reply:
Modified this function and `group_roles_update.sql` to use `user_id` instead of `uid`